pkg/client/record/event.go

```diff
@@ -111,19 +112,19 @@ func (eventBroadcaster *eventBroadcasterImpl) StartRecordingToSink(sink EventSin
 			// Events are safe to copy like this.
 			eventCopy := *event
 			event = &eventCopy
-
+			var patch []byte
 			previousEvent := eventCache.getEvent(event)
 			updateExistingEvent := previousEvent.Count > 0
 			if updateExistingEvent {
-				event.Count = previousEvent.Count + 1
-				event.FirstTimestamp = previousEvent.FirstTimestamp
+				// we still need to copy Name because the Patch relies on the Name to find the target event
 				event.Name = previousEvent.Name
-				event.ResourceVersion = previousEvent.ResourceVersion
+				time, _ := event.LastTimestamp.MarshalJSON()
+				patch = []byte(fmt.Sprintf("{\"count\":%d,\"lastTimestamp\":%s}", previousEvent.Count+1, string(time)))
```

Comment:
I think you are talking about this: https://github.com/kubernetes/kubernetes/blob/49d6c86eb764a8c825d78a7810f4d57db55e9308/pkg/util/strategicpatch/patch.go#L111

Let me try if I can use this to create patch for individual fields.